Dynatrace/dynatrace-operator
# Instrument CSI provisioner with Otel traces

## Description


With this otel is instrumented for the CSI provisioner. Following functions are traced:
- Reconcile
- getDynakube
- updateAgentInstallation
- installAgentZip
- collectGarbage

If you have feedback let me know, if those functions are fitting or if I should trace other functions.

[JIRA ticket](https://dt-rnd.atlassian.net/browse/K8S-9371)

## How can this be tested?


Deploy Operator, apply otel secret with necessary scopes and have a look at the distributed traces in your tenant. You should see aforementioned functions there.
